File: algorithms/tower_of_hanoi/javascript/tower_of_hanoi.js

// Prints the movement
function moveDisk(from_pole, to_pole) {
    console.log("Moving disk from " + 
        from_pole + " to " + to_pole);
}

// Tower of Hanoi implementation
function towerOfHanoi(height, from_pole=1, to_pole=2, aux=3) {
    if (height > 0) {
        towerOfHanoi(height - 1, from_pole, to_pole, aux);
        moveDisk(from_pole, to_pole);
        towerOfHanoi(height - 1, aux, from_pole, to_pole);
    }
}

towerOfHanoi(3);

File: algorithms/tower_of_hanoi/c/tower_of_hanoi.c
#include <stdio.h>

//Prints the movement
void moveDisk(from_pole, to_pole){
	printf("Moving disk from pole %d to pole %d\n", from_pole, to_pole);	
}

//Recursive implementation for Tower of Hanoi
void tower_of_hanoi(height, from_pole, to_pole, aux){
	if (height > 0){
		tower_of_hanoi(height-1, from_pole, to_pole, aux);
		moveDisk(from_pole, to_pole);
		tower_of_hanoi(height-1, aux, from_pole, to_pole);
	}
}

int main(){
	tower_of_hanoi(3);
}
